Lignite Coal

Danarca opened the pull request:
This has been slowly cooking since 1.6, so it have gone through 2 major compatches since its inception. I've done my best removing outdated references, and it seems to work as intended..

Crushita, the original modder for this project, did a bunch of research on lignite deposits, their locations and density afaik. His contributions in the map_data files are indicated with `# IMPERIA LIGNITE` above them.
He wasn't done though when he left, so I took over.

I've indicated the original amount of coal on the vanilla coal lines and the % that was moved to lignite.
Generally I've gone with 75% (50% some places) of the coal moved to lignite, with 75% of that lying undiscovered. The deposits are real-life accurate as best as I could, although I took data from the modern day... But I'm guessing locations of coal deposits doesn't change in 200 years.
Lignite Mining is also only 80% effective compared to normal coal mining, (considered when moving coal to lignite; 4 coal mines became 5 lignite) just to disincentive it, and outputs a whole lot more pollution. Which is accurate. 30 - 40% more pollution compared to coal (when measuring the energy output) is the general rule of thumb, apparently.

I haven't moved already-existing coal mines into lignite buildings, which is certainly a way this branch could be improved!

For now, I'm just gonna check on vic3-tiger to see if everything is ok.

Danarca committed:
Create babylon_mines.txt

Danarca committed:
Lignite PMs

Danarca committed:
Update 13_australasia.txt

Danarca committed:
Children yearn for the mines

Danarca committed:
Discovery Channel

Danarca committed:
Lignite mine in NSW now works :3

Crushita committed:
The Mines Yearn for the Children

Lignite mines added around the world. Mines and discovery event localized.

Crushita committed:
Re-encoded txt files to UTF-8 BOM

Danarca committed:
Merge branch 'Imperia-Core' into Imperia-Coal-Traits

Danarca committed:
Babylon -> Imperia renaming complete

Danarca committed:
Revival

Updated to 1.6
Imperia naming
Functionally complete

Danarca committed:
Vanilla mine sprites

Danarca committed:
Now appears on map

Danarca committed:
Merge branch 'Imperia-Core' into Imperia-Coal-Traits

Danarca committed:
Basic 1.8 compatch

Danarca committed:
Update basic building syntax

Danarca committed:
Lignite base pmg localisation drawn from coal mining loc

Danarca committed:
Lignite mine explosives pmg + loc

Danarca committed:
More pollution for lignite mining operations

Should really be applied where the lignite coal is burned, but... We only have 1 coal good type. This is the next best solution.

Danarca committed:
Some major lignite deposits added

Eastern Europe / Russia + Ethiopia + Costa Rica deposits added

Danarca committed:
Chinese and Indonesian lignite deposits added

Danarca committed:
Remove references to command building

Danarca committed:
Removal of outdated innovation multipliers

Danarca committed:
Reformatted all files to common sense

Danarca committed:
Ownership pmg removed

Danarca committed:
Lignite modifier_types localisation

Danarca committed:
Merge branch 'Imperia-Coal-Traits' of https://github.com/Imperia-Development-Victoria-3/Victoria-3-Dev into Imperia-Coal-Traits

HiddeLekanne commented on common/buildings/imperia_mines.txt:
```diff
@@ -0,0 +1,20 @@
+﻿building_lignite_mine = {
+	building_group = bg_lignite_mining
+	icon = "gfx/interface/icons/building_icons/coal_mine.dds"
```
New Icon needed?

Danarca replied:
I'll send a request to Sheep!

HiddeLekanne commented on common/production_methods/imperia_mines.txt:
```diff
@@ -0,0 +1,183 @@
+			building_employment_shopkeepers_add = 500
+			building_employment_laborers_add = 4500
+		}
+	}
```
Comparison to normal coalmine would be nice

Danarca replied:
There you go :)

HiddeLekanne commented on events/imperia_lignite_events.txt:
```diff
@@ -0,0 +1,49 @@
+			months = short_modifier_time
+		}
+	}
+	option = {
+		#the children yearn for the mines
+		name = lignite.1.b
+		trigger = {
+			owner = {
+				has_law = law_type:law_child_labor_allowed
+			}
+		}
+		add_modifier = {
+			name = state_lignite_rush
+			months = short_modifier_time
+		}
+	}
```
Shouldn't this have a different modifier?

Danarca replied:
It's just for a change in localisation, Crushita wanted a "The Children yearn for the mines" jokes in there :p

HiddeLekanne replied:
But wouldn't there be two options when triggered?

Danarca replied:
I thought for sure there was a trigger on the first option :s I'll add one, so only one of them pops

HiddeLekanne commented on common/buildings/imperia_mines.txt:
```diff
@@ -0,0 +1,20 @@
+﻿building_lignite_mine = {
```
This is the profitibilty graph compared to some other buildings:

![image](https://github.com/user-attachments/assets/ccc2be63-2acd-4ddc-a771-f8759c52e64b)

Check if that is desirable.

Danarca replied:
There's only 4 main production methods in the primary group though, whats that 5th one? Because that is definitely not intended :s

HiddeLekanne replied:
It's era level, not production method order.

Danarca replied:
Hmm, actually... If anything, 3 and 4 should be higher.. Didnt see 5 on the normal coal mine was that high as well. 
I.. Think we may have to go over the balancing collectively during the meeting :s

Danarca committed:
Comments on base output

Danarca committed:
Reformatted all files to common sense

Danarca committed:
Added trigger for base option in the lignite discovery event

Danarca committed:
Merge branch 'Imperia-Coal-Traits' of https://github.com/Imperia-Development-Victoria-3/Victoria-3-Dev into Imperia-Coal-Traits

Crushita commented:
THE CHILDREN YEARN FOR THE MINES, AND I HAVE GIVEN THEM THE MINES

Danarca committed:
Lignite to 70% input/output + steam donkey pm + icon

Danarca committed:
Vanilla indian city_types

Danarca committed:
Lignite (and charcoal to get of that error) meshes for India

Danarca committed:
Reformatted all files to common sense

Danarca commented:
Oh wait, Crushita had to do a once-over on any missing states for lignite, right? 
If thats the case, you can ignore my poke for a review for now Kikker :p 

I've also added the charcoal kiln to the india region, since I had to dig into the meshes anyways..

Danarca commented:
> Just to test out the copilot summary feature, Im gonna leave this here

This pull request introduces a new lignite mining feature, including the addition of a new building type, associated production methods, and modifiers. The changes span multiple files to incorporate lignite mining into the game, including new events and graphical updates for different city types.

### New Lignite Mining Feature:

* Added a new building group for lignite mining in `common/building_groups/imperia_building_groups.txt`, including properties such as `parent_group`, `default_building`, and `discoverable_resource`.
* Introduced a new building type for lignite mines in `common/buildings/imperia_mines.txt`, specifying attributes like `icon`, `city_type`, `levels_per_mesh`, and `production_method_groups`.
* Implemented new production methods for lignite mines in `common/production_methods/imperia_mines.txt`, detailing input/output values, state and building modifiers, and unlocking technologies.
* Added new modifiers for lignite mining in `common/modifier_type_definitions/imperia_modifier_types.txt`, including tax, mortality, standard of living, employee, and throughput modifiers.
* Created a new event for discovering lignite resources in `events/imperia_lignite_events.txt`, with triggers and options for adding a state modifier.

### Graphical Updates:

* Updated various city types to include the new lignite mine building mesh in files such as `gfx/map/city_data/city_types/african_mine.txt`, `gfx/map/city_data/city_types/american_mining_oilrig.txt`, `gfx/map/city_data/city_types/arabic_mine.txt`, `gfx/map/city_data/city_types/asian_mine.txt`, and `gfx/map/city_data/city_types/default_mine.txt`. [[1]](diffhunk://#diff-401c8eb9f2235b8d9ee5bd668d332c49bd29cf8179e79cb4f945113a4c98f22eR1-R69) [[2]](diffhunk://#diff-f530c4f8579454f52cab429ac0d5609e3f14761a2ed616b4adea447bda886e21R1-R89) [[3]](diffhunk://#diff-7daf65cef6e90346b0194a4d9b5413a93e7c048fc1ef22cd5d7ad056addbdbd9R1-R69) [[4]](diffhunk://#diff-b94636b9815782e38851b9986425b4b7a207eb6ede2f0a0df9e7d93e09528b50R1-R69) [[5]](diffhunk://#diff-90b4a929175388bfbc495de98ef9620d16c3ad7e30467f692bf1eb6c84d24a9bR1-R70)

### Historical and On-Action Changes:

* Added historical data for lignite mines in `common/history/buildings/lignite/13_australasia.txt`, creating a building in New South Wales.
* Introduced new on-action triggers for resource discovery in `common/on_actions/imperia_index_on_actions.txt` and `common/on_actions/imperia_misc_on_actions.txt`. [[1]](diffhunk://#diff-82483ff9c53f35c41e28fc3126a38a9f4bac162822905df726c73ce6c975b4f5R58-R64) [[2]](diffhunk://#diff-37ae6caa09be13fc5b45649308ac0a11e2b32ca3e12d88b86359a7033d314a24R1-R6)

### Static Modifiers:

* Added a new state modifier for a lignite rush in `common/static_modifiers/imperia_modifiers.txt`, affecting employee multiplier and migration pull.

HiddeLekanne commented:
Still profitibility:

![image](https://github.com/user-attachments/assets/73980527-2971-4a2b-b84d-5a46714debf5)

HiddeLekanne commented:
![image](https://github.com/user-attachments/assets/e7dbd574-b95d-468c-8ca5-0513065e82e8)

HiddeLekanne commented:
So if you could still doe the transport pm that would be great.

Danarca committed:
Lignite-specific train automation PMs

Danarca committed:
Reformatted all files to common sense
